### PR TITLE
[Paddle-pipelines] Add ReAct zh examples

### PR DESCRIPTION
**Pull request by w5688414**
### PR types

+ New features

### PR changes

+ APIs

### Description

+ 示例

```
Agent few-shot-react started with {'query': '成龙的儿子的年龄是多少?', 'params': None}
```

```
让我们一步一步地思考。要回答这个问题，我们需要了解成龙是谁，他的儿子叫什么名字以及他的年龄。

工具: 搜索
工具输入: 成龙是谁？
观察: 成龙是一位著名的中国武打演员和电影导演。

工具: 搜索
工具输入: 成龙的儿子叫什么名字？
观察: 成龙的儿子名叫房祖名。

工具: 搜索
工具输入: 房祖名的年龄是多少？
观察: 房祖名出生于1982年，今年大约40岁。

思考: 根据搜索结果，成龙的儿子的年龄是大约40岁。所以最终答案是大约40岁。
最终答案: 大约40岁
```

```
让我们一步一步地思考。要回答这个问题，我们首先需要了解三体是什么。
工具: 搜索
工具输入: 三体是什么？

观察: 三体是离太阳系最近的恒星系，在这里诞生了三体和三体文明。
思考: 我们需要找到三体的作者。
工具: 搜索
工具输入: 三体的作者是谁？


观察: 刘慈欣 (分数: 0.33)
思考: 根据搜索结果，三体的作者是刘慈欣。所以最终答案是刘慈欣。最终答案: 刘慈欣
```

+ 本地化语义检索+ ReAct

```
让我们一步一步地思考。要回答这个问题，我们首先需要了解范冰冰是谁。
工具: 搜索
工具输入: 范冰冰是谁？

观察: 范冰冰，1981年9月16日出生于山东青岛，华语影视女演员、制片人、歌手2019年10月，范冰冰入选《福布斯》全球100位最有权势女性榜单。
思考: 根据搜索结果，范冰冰是一位华语影视女演员、制片人、歌手。现在我们需要找到她的身高是多少。
工具: 搜索
工具输入: 范冰冰的身高是多少？

观察: 范冰冰的真实身高是160-162厘米。
思考: 根据搜索结果，范冰冰的真实身高是160-162厘米。所以最终答案是160-162厘米。最终答案: 160-162厘米
```
```
让我们一步一步地思考。要回答这个问题，我们首先需要了解武则天是谁。
工具: 搜索
工具输入: 武则天是谁？

观察: 武则天是唐朝的女皇帝。
思考: 我们现在知道武则天是唐朝的女皇帝。接下来，我们需要知道她把皇位传给了谁。
工具: 搜索
工具输入: 武则天传位给了谁？

观察: 武则天传位给了李旦。
思考: 根据搜索结果，武则天传位给了李旦。所以最终答案是李旦。
最终答案: 李旦
```

TODO:
- [x]  Fix Agents with no stop sign 
- [x] Fix dead loop bug
- [x]  Add Agents with semantic search examples
- [x] Add Unitest

### CODE REVIEW

File: pipelines/pipelines/agents/base.py [hunk 1]
```diff
@@ -193,10 +193,16 @@ def run_tool(self, llm_response: str, params: Optional[Dict[str, Any]] = None) -
...
+                        observation_prefix="观察: ",
+                        llm_prefix="思考: ",
                         color=tool.logging_color,
                     )
+                    # self.callback_manager.on_tool_finish(
+                    #     tool_result,
+                    #     observation_prefix="Observation: ",
+                    #     llm_prefix="Thought: ",
+                    #     color=tool.logging_color,
+                    # )
```
Comment (sijunhe) [inline pipelines/pipelines/agents/base.py]: 这里不能hardcode, 应该通过Agent传参

File: pipelines/pipelines/agents/agent_step.py [hunk 2]
```diff
@@ -118,10 +118,13 @@ def completed(self, observation: Optional[str]) -> None:
...
+        #     f"{self.prompt_node_response}\nObservation: {observation}\nThought:"
+        #     if observation
+        #     else self.prompt_node_response
+        # )
         self.transcript += (
-            f"{self.prompt_node_response}\nObservation: {observation}\nThought:"
-            if observation
-            else self.prompt_node_response
+            f"{self.prompt_node_response}\n观察: {observation}\n思考: " if observation else self.prompt_node_response
         )
```
Comment (sijunhe) [inline pipelines/pipelines/agents/agent_step.py]: 这里不能hardcode, 应该通过Agent传参

File: pipelines/pipelines/agents/base.py (hunk 1 above)
Comment (w5688414) [reply pipelines/pipelines/agents/base.py]: 已修改

File: pipelines/pipelines/agents/agent_step.py (hunk 2 above)
Comment (w5688414) [reply pipelines/pipelines/agents/agent_step.py]: 已修改

**Review by sijunhe — APPROVED** [review]
lgtm

**Pull request merged**